tests/extension/oneapi_sub_group_mask/sub_group_mask_flip_api.cpp

```diff
@@ -18,11 +18,16 @@ using namespace sycl_cts;
 struct check_result_flip {
   bool operator()(sycl::ext::oneapi::sub_group_mask &sub_group_mask,
                   const sycl::sub_group &sub_group) {
-    unsigned long before_flip, after_flip;
-    sub_group_mask.extract_bits(before_flip);
+    // since all bits need to be checked and size of sub_group_mask is not
+    // known beforehand we can't use fixed type to extract bits and check them
+    // after the flip
```

Comment:
I disagree with the statement and suggest removing or adjusting this comment.
NOTE: there is no such a comment in tests/extension/oneapi_sub_group_mask/sub_group_mask_set_api.cpp

I think something like this should work:
```c++
    unsigned long before_flip, after_flip;
    sub_group_mask.extract_bits(before_flip);
    sub_group_mask.flip();
    sub_group_mask.extract_bits(after_flip);
    // mask off irrelevant bits
    unsigned long mask = ULONG_MAX << sub_group_mask.size();
    after_flip = after_flip & mask;
    before_flip = before_flip | ~mask;
    return after_flip == ~before_flip;
```
`unsigned long` size is not "fixed", but the same can be done with any fixed type as long as it can hold all `sub_group_mask` bits.

I guess the performance of this part is not critical, so I'm okay to have a loop as it's easier to read.